spec/teacher_spec.rb

```diff
@@ -0,0 +1,13 @@
+require_relative '../teacher'
+require 'json'
+
+describe Teacher do
+  before :each do
+    @teacher = Teacher.new 45, 'Biology'
+  end
+  context 'When testing the Teacher class' do
+    it 'Given age, and specialization teacher object must be returned' do
+      expect(@teacher).to be_an_instance_of Teacher
+    end
+  end
+end
```

Comment:
- [ ] Good job, but some tests are missing, you are missing tests to check if `The constructor method sets the default value for the optional name parameter if it is not provided and also tests for the method `can_use_services?`